### PR TITLE
[BUG]  An alias "filerename" was requested but no service could be found  ZF2.5.3

### PR DESCRIPTION
**Pull request by buse974**
**Bug with zend-filter 2.6.0** (ZF2.5.3) 

> An alias "filerename" was requested but no service could be found.

Since version 2.6.0 of zend-filter the aliases property is used in Zend\Filter\FilterPluginManager. I merged aliases property in the constructor Zend\File\Transfer\Adapter\FilterPluginManager. 

_Here is a minimal example of the problem I ran into._

``` php
$adp = new Http();
$adp->addFilter('Rename', array('target' => $nameMod.'.dat'));
```

it works and tests pass.

### CODE REVIEW

**Comment by remicollet** [comment]
This PR fix test suite with recent component.

See (Fedora PHP Stack CI) https://apps.fedoraproject.org/koschei/package/php-zendframework-zend-file

File: test/Transfer/Adapter/HttpTestMockAdapter.php [hunk 1]
```diff
@@ -18,6 +18,8 @@
  */
 class HttpTestMockAdapter extends Adapter\Http
 {
+    static $aa = true;
```
Comment (Ocramius) [inline test/Transfer/Adapter/HttpTestMockAdapter.php:21]: Can this be given a better name?

File: src/Transfer/Adapter/FilterPluginManager.php [hunk 2]
```diff
@@ -20,16 +21,19 @@
...
+            'decrypt'       => File\Decrypt::class,
+            'encrypt'       => File\Encrypt::class,
+            'lowercase'     => File\LowerCase::class,
+            'rename'        => File\Rename::class,
+            'uppercase'     => File\UpperCase::class
+        ), $this->aliases);
+    }
+
 }
+
```
Comment (Ocramius) [inline src/Transfer/Adapter/FilterPluginManager.php:39]: revert

File: src/Transfer/Adapter/FilterPluginManager.php [hunk 3]
```diff
@@ -20,16 +21,19 @@
  */
 class FilterPluginManager extends BaseManager
 {
-    /**
-     * Default set of filters
-     *
-     * @var array
-     */
-    protected $aliases = [
```
Comment (Ocramius) [inline src/Transfer/Adapter/FilterPluginManager.php:28]: This is a BC break

File: src/Transfer/Adapter/FilterPluginManager.php [hunk 4]
```diff
@@ -20,16 +21,19 @@
...
-        'lowercase' => 'filelowercase',
-        'rename'    => 'filerename',
-        'uppercase' => 'fileuppercase',
-    ];
+
+    public function __construct($configOrContainerInstance = null, array $v3config = []) 
+    {
+	parent::__construct($configOrContainerInstance, $v3config);
+        
+	$this->aliases = array_merge(array(
```
Comment (Ocramius) [inline src/Transfer/Adapter/FilterPluginManager.php:29]: You can use the short array syntax. Remember that multi-line parameters should stay on a line of their own as per PSR-7

File: src/Transfer/Adapter/FilterPluginManager.php [hunk 5]
```diff
@@ -20,16 +21,19 @@
...
-     */
-    protected $aliases = [
-        'decrypt'   => 'filedecrypt',
-        'encrypt'   => 'fileencrypt',
-        'lowercase' => 'filelowercase',
-        'rename'    => 'filerename',
-        'uppercase' => 'fileuppercase',
-    ];
+
+    public function __construct($configOrContainerInstance = null, array $v3config = [])
```
Comment (Ocramius) [inline src/Transfer/Adapter/FilterPluginManager.php:25]: Constructor needs documentation docblock

Comment (Ocramius) [inline src/Transfer/Adapter/FilterPluginManager.php:25]: `v3config` naming is a bit confusing

File: test/Transfer/Adapter/HttpTestMockAdapter.php [hunk 6]
```diff
@@ -50,7 +52,7 @@ public static function uPTest($id)
     }
 
     public function switchApcToUP()
-    {
+    {	static::$aa = false;
```
Comment (Ocramius) [inline test/Transfer/Adapter/HttpTestMockAdapter.php:55]: Should be on a new line. Also: why is this change required?

File: test/Transfer/Adapter/AbstractTest.php [hunk 7]
```diff
@@ -301,7 +301,7 @@ public function testGetFilterShouldReturnNullWhenNoMatchingIdentifierExists()
 
     public function testAdapterShouldAllowPullingFiltersByFile()
     {
-        $this->adapter->addFilter('Boolean', 1, 'foo');
+        $this->adapter->addFilter('Boolean', [1], 'foo');
```
Comment (Ocramius) [inline test/Transfer/Adapter/AbstractTest.php:304]: This seems like a BC break

File: composer.json [hunk 8]
```diff
@@ -17,10 +17,12 @@
         "zendframework/zend-stdlib": "~2.5"
     },
     "require-dev": {
-        "zendframework/zend-filter": "~2.5",
+        "zendframework/zend-filter": "~2.6",
         "zendframework/zend-i18n": "~2.5",
         "zendframework/zend-servicemanager": "~2.5",
         "zendframework/zend-validator": "~2.5",
+        "zendframework/zend-progressbar": "~2.5",
```
Comment (Ocramius) [inline composer.json:24]: Needed? In which case?

File: composer.json [hunk 9]
```diff
@@ -17,10 +17,12 @@
...
+        "zendframework/zend-session": "~2.5",
```
Comment (Ocramius) [inline composer.json:25]: Needed? In which case?

File: src/Transfer/Adapter/FilterPluginManager.php (hunk 4 above)
Comment (Maks3w) [inline src/Transfer/Adapter/FilterPluginManager.php:29]: @Ocramius wants to say PSR-2 :)

Comment (Ocramius) [inline src/Transfer/Adapter/FilterPluginManager.php:29]: Long night :P

File: composer.json (hunk 8 above)
Comment (weierophinney) [inline composer.json:24]: Interestingly, in `ZendTest\File\Transfer\Adatper\HttpTest`. Missing requirement, evidently.

Comment (weierophinney) [inline composer.json:24]: (and that test requires it in order to test `Zend\File\Transfer\Adapter\Http`)

File: composer.json (hunk 9 above)
Comment (weierophinney) [inline composer.json:25]: Yeah, I cannot find a requirement for this anywhere.

File: src/Transfer/Adapter/FilterPluginManager.php (hunk 3 above)
Comment (weierophinney) [inline src/Transfer/Adapter/FilterPluginManager.php:28]: @Ocramius yes and no.

Prior to zend-filter 2.6.0, `Zend\Filter\FilterPluginManager` did not define any `$aliases`, which made adding them to this extension safe. However, part of the 2.6.0 update was separating invokables into aliases + factories. That change has made it so that adding the aliases property in this extension breaks expectations, as other filters expected to be present are missing. The approach being done in this patch is to merge this "default set" with those defined in the parent, via the constructor, which is both backwards and forwards compatible.

File: src/Transfer/Adapter/FilterPluginManager.php (hunk 5 above)
Comment (weierophinney) [inline src/Transfer/Adapter/FilterPluginManager.php:25]: @Ocramius that's the naming current under zend-servicemanager; in v3, it's a configuration array; v2 doesn't accept a second argument. Hence, `$v3config`. (Hey, it's not great, but I got a green light on review for that...)

File: test/Transfer/Adapter/HttpTestMockAdapter.php (hunk 6 above)
Comment (weierophinney) [inline test/Transfer/Adapter/HttpTestMockAdapter.php:55]: I've pulled the changes in locally, and none of the changes to this particular class are necessary; tests pass with and without them.

File: test/Transfer/Adapter/AbstractTest.php (hunk 7 above)
Comment (weierophinney) [inline test/Transfer/Adapter/AbstractTest.php:304]: On investigation, this is due to how "creation options" are handled in v2 with the invokable factory. I think there may be a way to work around it; testing something now.

File: src/Transfer/Adapter/FilterPluginManager.php (hunk 3 above)
Comment (Ocramius) [inline src/Transfer/Adapter/FilterPluginManager.php:28]: Still a BC break from a consumer of this library perspective though :-\

File: src/Transfer/Adapter/FilterPluginManager.php (hunk 5 above)
Comment (Ocramius) [inline src/Transfer/Adapter/FilterPluginManager.php:25]: > I got a green light on review for that

@weierophinney that's ok, didn't know :-)
